### ISSUE TITLE
'Expiration date cannot exceed x days' message shows wrong number of days

### ISSUE THREAD

**Issue by HanaGemela** [owncloud/client#7328]
Client: 2.6.0alpha2 (build 12128)
macOS: 10.14.5
Server:10.2.1. (stable)

Steps to recreate:
1) On the server go to Settings -> Admin -> Sharing
2) Set 'Expire after 10 days'
3) Tick Enforce expiration date
4) On the server go to Settings -> Admin -> Security (Password policy app has to be installed)
5) Set '2 days until link expires if password is not set'
6) In the client create a new public link without a password and with expiration date 11 days in the future

Actual result: Error message that the expiration date cannot be more than 10 days in the future.

7) Edit the expiration date in the client and set it to 9 days 

Actual result: Error message that the expiration date cannot be more than 2 days in the future.
Basically user got two error messages. The first one told him that the date has to be 10 days in the future, once he corrected the date to be 9 days in the future, he gets another message saying it has to be two days.

Expected result: Just one error message showing the lower number of days in the future required to create a public link

Note: The numbers here are just an example

**Comment by ckamm** [comment]
The client reports the error message that's sent from the server. Also, it can't detect the Security settings so there's no way to supplement a better error message. (if it could detect it, it would just make invalid inputs impossible)

**Comment by ckamm** [comment]
I think this is mitigated by the client not allowing users to perform the step

> 6. In the client create a new public link without a password and with expiration date 11 days in the future

When a maximum expiration date 10 days into the future is set in the client. Did you restart the client after the server settings change?

**Comment by HanaGemela** [comment]
Cannot recreate anymore. I think it might have been caused by wrong time on the server. I'll close this issue but will keep an eye on it
